### PR TITLE
Barista/generate strapicms pages

### PR DESCRIPTION
**Pull request by lara-aigmueller**
### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to the Dynatrace Angular Component library (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please fill out the checkbox below: <br>

#### Type of PR

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or change that would cause existing functionality to
      not work as expected)
- [ ] Documentation update (changes to documentation)
- [ ] Other (if none of the above apply)

#### Checklist

- [ ] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: tools/barista/src/types.ts [hunk 1]
```diff
@@ -55,3 +55,50 @@ export interface BaContributor {
...
+  name: string;
+}
+
+/** Strapi page category */
+interface BaStrapiCategory extends BaStrapiBase {
+  title: string;
+}
+
+/** Strapi contributor (UX/Dev support) */
+interface BaStrapiContributor extends BaStrapiNamedEntity {
```
Comment (thomaspink) [inline tools/barista/src/types.ts]: should be exported

File: tools/barista/src/types.ts [hunk 2]
```diff
@@ -55,3 +55,50 @@ export interface BaContributor {
...
+  updated_at: number;
+}
+
+/** Strapi content type with a name */
+interface BaStrapiNamedEntity extends BaStrapiBase {
+  name: string;
+}
+
+/** Strapi page category */
+interface BaStrapiCategory extends BaStrapiBase {
```
Comment (thomaspink) [inline tools/barista/src/types.ts]: should be exported

File: tools/barista/src/types.ts [hunk 3]
```diff
@@ -55,3 +55,50 @@ export interface BaContributor {
...
+
+/** Base interface for Strapi content types */
+interface BaStrapiBase {
+  id: number;
+  created_at: number;
+  updated_at: number;
+}
+
+/** Strapi content type with a name */
+interface BaStrapiNamedEntity extends BaStrapiBase {
```
Comment (thomaspink) [inline tools/barista/src/types.ts]: should be exported

File: tools/barista/src/types.ts [hunk 4]
```diff
@@ -55,3 +55,50 @@ export interface BaContributor {
   name: string;
   gitHubUser: string;
 }
+
+/** Base interface for Strapi content types */
+interface BaStrapiBase {
```
Comment (thomaspink) [inline tools/barista/src/types.ts]: should be exported

**Comment by lara-aigmueller** [comment]
Added a fallback if no CMS endpoint is given (https://github.com/dynatrace-oss/barista/pull/199/files#diff-d9f0f825ed72dbedfe1819830acf4742R44).

**Comment by thomaspink** [comment]
LGTM!

File: libs/shared/barista-definitions/src/lib/barista-definitions.ts [hunk 5]
```diff
@@ -67,3 +67,50 @@ export interface BaContributor {
...
+
+/** Strapi snippet */
+export interface BaStrapiSnippet extends BaStrapiBase {
+  slotID: string;
+  title: string;
+  content: string;
+}
+
+/** Strapi content types */
+export enum StrapiContentType {
```
Comment (tomheller) [inline libs/shared/barista-definitions/src/lib/barista-definitions.ts]: Please prefix with `Ba` when it is exported.

File: tools/barista/src/utils/fetch-strapi-content.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+ * Fetches data from Strapi CMS based on the given
+ * requestPath.
+ *
+ * Read more about the Strapi Content API here
+ * https://strapi.io/documentation/3.0.0-beta.x/content-api/api-endpoints.html
+ */
+async function fetchContent(requestPath: string): Promise<any> {
+  const options = {
+    hostname: STRAPI_ENDPOINT,
+    port: 5100,
```
Comment (tomheller) [inline tools/barista/src/utils/fetch-strapi-content.ts]: We could move the port also to an env variable, but this is not considered blocking for me.

**Pull request merged**